services/blockstorage/sync/state_processing_blocks.go

```diff
@@ -0,0 +1,69 @@
+package sync
+
+import (
+	"context"
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+)
+
+type processingBlocksState struct {
+	blocks  *gossipmessages.BlockSyncResponseMessage
+	logger  log.BasicLogger
+	storage BlockSyncStorage
+	sf      *stateFactory
+}
+
+func (s *processingBlocksState) name() string {
+	return "processing-blocks-state"
+}
+
+func (s *processingBlocksState) processState(ctx context.Context) syncState {
+	if ctx.Err() == context.Canceled {
+		return nil
+	}
+
+	if s.blocks == nil {
+		s.logger.Error("fatal error in block sync, received no blocks to processing blocks state")
```

Comment:
well, *fatal* might be a bit of an exaggeration here, no? I think this is one of those `throw new ThisShouldNeverHappenException()` scenarios

Reply:
by which I mean, "unexpected error in block sync"

Reply:
correct, too much drama